contracts/Stage3/Directory.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract StakingDirectory is Ownable {
+
+    address[] public entries;
+
+    constructor() {}
+
+    function add(address _address) external onlyOwner {
+        entries.push(_address);
+    }
+
+    function deleteEntry(uint256 index) private {
+        for (uint256 i = index; i < entries.length - 1; i++) entries[i] = entries[i+1];
+        entries.pop();
+    }
+
+    function del(address _address) external onlyOwner {
```

Comment:
assume `entries = [0x1, 0x2, 0x3, 0x4]`.

you call `del(0x2)` and the expected outcome is `entries = [0x1, 0x3, 0x4]`.

instead it will do `deleteEntry(1)` and do:
1) `entries[1] = entries[2]` which results in `entries = [0x1, 0x3, 0x3, 0x4]`.
2) `entries.pop()` which results in `entries = [0x1, 0x3, 0x3]`

which is not the same as the expected